File: EdouardMathieu/find-reusable-blocks/find-reusable-blocks.R
# Change this to the name of the topic page
PAGE <- "Smoking"

# Install data.table package if not already installed
if (!require("data.table")) install.packages("data.table")
library(data.table)

url <- "https://docs.google.com/spreadsheets/d/e/2PACX-1vR1Tylsvt9CQ5v5RsxXG0FuL_DdnMAdUWG7cSx6yY1aiVddg3w9YYQXdTCRxx62k7gg3VBYoMeShHHB/pub?gid=598437417&single=true&output=csv"
df <- fread(url, showProgress = F)
df <- df[post_status == "publish"]

topic_page <- df[post_title == PAGE]

cat("Blocks used in the", PAGE, "topic page:\n")
print(topic_page[, c("reusable_block_slug", "reusable_block_title")])

blocks_used <- topic_page$reusable_block_slug
other_posts <- df[reusable_block_slug %in% blocks_used & post_title != PAGE]

cat("\nThese blocks are also used in:\n")
print(other_posts[, c("reusable_block_title", "post_title", "post_slug")])

File: EdouardMathieu/find-reusable-blocks/find-reusable-blocks.py
import argparse
import pandas as pd

# Parse command-line arguments
parser = argparse.ArgumentParser()
parser.add_argument("--slug", help="Name of the topic page")
args = parser.parse_args()

# Set the SLUG based on the command-line argument
# If missing, raise an error
if args.slug is None:
    raise ValueError("Please provide a topic page name")
SLUG = args.slug

# URL to fetch data from
url = "https://docs.google.com/spreadsheets/d/e/2PACX-1vR1Tylsvt9CQ5v5RsxXG0FuL_DdnMAdUWG7cSx6yY1aiVddg3w9YYQXdTCRxx62k7gg3VBYoMeShHHB/pub?gid=598437417&single=true&output=csv"

# Load data into a DataFrame
df = pd.read_csv(url)

# Filter rows where post_status is 'publish'
df = df[df["post_status"] == "publish"]

# Get the rows for the specified topic page
topic_page = df[df["post_slug"] == SLUG]

print("Blocks used in the", SLUG, "topic page:\n")
print(topic_page[["reusable_block_slug", "reusable_block_title"]])

blocks_used = topic_page["reusable_block_slug"].tolist()
other_posts = df[
    df["reusable_block_slug"].isin(blocks_used) & (df["post_slug"] != SLUG)
]

print("\nThese blocks are also used in:\n")
print(other_posts[["reusable_block_title", "post_title", "post_slug"]])
